src/main/java/org/zalando/nakadi/service/subscription/SubscriptionService.java

```diff
@@ -180,7 +184,15 @@ public UriComponents getSubscriptionUri(final Subscription subscription) {
         final Set<String> eventTypesFilter = eventTypes == null ? ImmutableSet.of() : eventTypes;
         final Optional<String> owningAppOption = Optional.ofNullable(owningApplication);
         final List<Subscription> subscriptions =
-                subscriptionRepository.listSubscriptions(eventTypesFilter, owningAppOption, offset, limit);
+                authorizationService
+                        .filter(subscriptionRepository
+                                .listSubscriptions(eventTypesFilter, owningAppOption, offset, limit)
+                                .stream()
+                                .map(Subscription::asResource)
+                                .collect(Collectors.toList()))
+                        .stream()
+                        .map(Resource<Subscription>::get)
+                        .collect(Collectors.toList());
```

Comment:
Maybe we need to run this several times until we accumulated the requested number of items in the page?